routes/decoratorRequestBody_test.go

```diff
@@ -0,0 +1,97 @@
+package routes
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+)
+
+type bodyEchoTest struct {
+	Foo string `json:"foo" req:"nonzero"`
+	Bar int    `json:"bar"`
+}
+
+func bodyEchoHandler(r *http.Request, a Arguments) (int, interface{}) {
+	var body bodyEchoTest
+	MustRequestBody(a, &body)
+	return http.StatusOK, body
+}
+
+func TestEchoHandler(t *testing.T) {
+	example := bodyEchoTest{"test", 42}
+	arguments := make(Arguments)
+	arguments[argumentKeyBody] = reflect.ValueOf(example)
+	status, response := bodyEchoHandler(nil, arguments)
+	if status != http.StatusOK {
+		t.Errorf("Status code should be %d, is %d.", http.StatusOK, status)
+	}
+	if response != example {
+		t.Errorf("Response should be %v, is %v.", example, response)
+	}
+}
+
+func TestSuccessfulValidation(t *testing.T) {
+	example := bodyEchoTest{"test", 42}
+	handler := H(bodyEchoHandler).With(
+		RequestBody{example},
+	)
+	bs, _ := json.Marshal(example)
```

Comment:
Why don't you check for the `error` return of `json.Marshal` ?

Reply:
Because with `example` being a very well-defined structure which `encoding/json` will not fail with, it would only make the _test_ harder to read, with no gain.